lib/scale.js

```diff
@@ -102,6 +102,32 @@ Scale.prototype = {
     this.tonic = scale.tonic;
 
     return this;
+  },
+
+  /**
+   * Get the sequence of notes representing key signature of a scale, in proper order.
+   * There are two possible signatures for any scale: flats and sharps.
+   * Sharps signature is got by traversing notes by P5 intervals to the right
+   * till all the notes of the scale are covered.
+   * Flats signature is got by traversing by P4 intervals to the left in the same manner, which is equal to reversed sharps.
+   */
+  signature: function() {
+    var sequence = this.notes().sort(function (a, b) {
+      var aCoord = a.coord[1] < 0 ? a.coord[1] + 12 : a.coord[1];
+      var bCoord = b.coord[1] < 0 ? b.coord[1] + 12 : b.coord[1];
+      return aCoord - bCoord;
+    });
+
+    //if at least one flat - return flats signature
+    if (sequence.some(function (note) {
```

Comment:
If you could maybe break this into two lines (a bit more descriptive):

    var hasFlat = sequence.some(function (note) {
      return note.accidentalValue() < 0;
    });
    
    if (hasFlat) { ... }